Framework/API/src/ExperimentInfo.cpp

```diff
@@ -1115,17 +1118,11 @@ ExperimentInfo::getInstrumentFilename(const std::string &instrumentName,
 
 /** Return a const reference to the DetectorInfo object.
  *
- * Any modifications of the instrument or instrument parameters will invalidate
+ * Setting a new instrument via ExperimentInfo::setInstrument will invalidate
```

Comment:
Just a question: wouldn't this be a legitimate use case for `std::weak_ptr` or similar?